src/pages-and-resources/progress/Settings.jsx

```diff
@@ -0,0 +1,50 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+
+import * as Yup from 'yup';
+
+import { injectIntl, intlShape } from '@edx/frontend-platform/i18n';
+
+import FormSwitchGroup from '../../generic/FormSwitchGroup';
+import AppSettingsModal from '../app-settings-modal/AppSettingsModal';
+import messages from './messages';
+
+function ProgressSettings({ intl, onClose }) {
+  // TODO: Save settings using APIs here
+  const handleSettingsSave = () => null;
+  return (
+    <AppSettingsModal
+      appId="progress"
+      title={intl.formatMessage(messages.heading)}
+      enableAppHelp={intl.formatMessage(messages.enableProgressHelp)}
+      enableAppLabel={intl.formatMessage(messages.enableProgressLabel)}
+      learnMoreText={intl.formatMessage(messages.enableProgressLink)}
+      onClose={onClose}
+      // TODO: fetch these values from the redux store, via APIs
+      initialValues={{ enableProgressGraph: false }}
+      validationSchema={{ enableProgressGraph: Yup.boolean() }}
```

Comment:
@xitij2000, I'm getting error:
```
Warning: Failed prop type: Invalid prop `validationSchema.enableProgressGraph` of type `object` supplied to `AppSettingsModal`, expected `function`.
```

Looks like prop types for `AppSettingsModal` should be updated?

Reply:
Updated!